Include/internal/pycore_signal.h

```diff
@@ -38,12 +37,12 @@ PyAPI_FUNC(void) _Py_RestoreSignals(void);
 #define INVALID_FD (-1)
 
 struct _signals_runtime_state {
-    volatile struct {
-        _Py_atomic_int tripped;
+    struct {
+        int tripped;
         /* func is atomic to ensure that PyErr_SetInterrupt is async-signal-safe
          * (even though it would probably be otherwise, anyway).
          */
-        _Py_atomic_address func;
+        uintptr_t func;
```

Comment:
It looks like `func` is always `PyObject*`. It might be cleaner to declare it as such and use `_Py_atomic_load_ptr`/`_Py_atomic_store_ptr` instead of operating on `uintptr_t`.

Reply:
Makes sense!